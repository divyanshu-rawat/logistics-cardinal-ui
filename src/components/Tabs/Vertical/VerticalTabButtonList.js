import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { withTheme } from 'styled-components';
import { Fade } from 'react-reveal';

import VerticalTabButtonStyled from './VerticalButtonStyled';
import Icon from '../../Icons';
import Text from '../../Text';

export const VerticalTabButtonList = ({
  options,
  activeTab,
  onClick,
  theme,
}) => (
  <Fragment>
    {options.map((item, key) => (
      <VerticalTabButtonStyled
        key={`ComposedVerticalTabButton-${item}`}
        isActive={activeTab === key}
        onClick={onClick(key)}
      >
        <Text
          margin="0"
          style={{ flex: 1, textDecoration: 'none' }}
          truncate
          color={
            activeTab === key
              ? theme.verticalTabButtonActiveTextColor
              : theme.verticalTabButtonTextColor
          }
        >
          {item}
        </Text>
        <Fade left duration={300} when={activeTab === key}>
          <span>
            <Icon.ArrowRight />
          </span>
        </Fade>
      </VerticalTabButtonStyled>
    ))}
  </Fragment>
);

VerticalTabButtonList.propTypes = {
  options: PropTypes.array.isRequired,
  activeTab: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
  theme: PropTypes.object.isRequired,
};

export default withTheme(VerticalTabButtonList);
