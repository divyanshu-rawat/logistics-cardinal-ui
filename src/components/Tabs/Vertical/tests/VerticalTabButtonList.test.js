import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { VerticalTabButtonList } from '../VerticalTabButtonList';

const renderComponent = () => (
  <VerticalTabButtonList
    activeTab={0}
    onClick={jest.fn()}
    options={['Forecast', 'History']}
    theme={{ colors: { primary100: '' } }}
  />
);

describe('<VerticalTabButtonList />', () => {
  it('should render the component', () => {
    const tree = toJson(shallow(renderComponent()));
    expect(tree).toMatchSnapshot();
  });
});
