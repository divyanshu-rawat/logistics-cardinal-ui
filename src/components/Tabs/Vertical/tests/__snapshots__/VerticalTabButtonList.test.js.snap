// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerticalTabButtonList /> should render the component 1`] = `
<Fragment>
  <VerticalButtonStyled__VerticalTabButtonStyled
    isActive={true}
    key="ComposedVerticalTabButton-Forecast"
  >
    <Text
      as="span"
      margin="0"
      style={
        Object {
          "flex": 1,
          "textDecoration": "none",
        }
      }
      truncate={true}
    >
      Forecast
    </Text>
    <Fade
      duration={300}
      left={true}
      when={true}
    >
      <span>
        <ArrowRight
          height="6"
          viewBox="0 0 13 6"
          width="13"
          xmlns="http://www.w3.org/2000/svg"
        />
      </span>
    </Fade>
  </VerticalButtonStyled__VerticalTabButtonStyled>
  <VerticalButtonStyled__VerticalTabButtonStyled
    isActive={false}
    key="ComposedVerticalTabButton-History"
  >
    <Text
      as="span"
      margin="0"
      style={
        Object {
          "flex": 1,
          "textDecoration": "none",
        }
      }
      truncate={true}
    >
      History
    </Text>
    <Fade
      duration={300}
      left={true}
      when={false}
    >
      <span>
        <ArrowRight
          height="6"
          viewBox="0 0 13 6"
          width="13"
          xmlns="http://www.w3.org/2000/svg"
        />
      </span>
    </Fade>
  </VerticalButtonStyled__VerticalTabButtonStyled>
</Fragment>
`;
